### PR TITLE
[filesystem/samba] - support for non-default smb ports

### PR DESCRIPTION
**Pull request by mariuscampan**
http://trac.kodi.tv/ticket/16514

### CODE REVIEW

**Comment by MrMC** [comment]
tested or assumed ?

**Comment by mariuscampan** [comment]
Tested on Ubuntu 15.10 with Kodi branch Jarvis. I'm in the process of build master+this patch for android - might take a while as it seems that the build process for android is not as straightforward as the Linux one :). I will post an update once I have the build.

**Comment by MrMC** [comment]
very good then :) 

Note that this will silently fail on the samba-gplv2 flavor. source code does not support custom port.

**Comment by mariuscampan** [comment]
For android, the patch doesn't work as samba 3.6.12 that is used doesn't support custom port. samba 4.1.0 stable release seems to be the first version supporting it.

Well, at least this patch allows kodi to do the right thing in not stripping the port off.

**Comment by MrMC** [comment]
Android used the samba-gplv2 flavor :) The port commit could be patched in to it.

**Comment by mariuscampan** [comment]
On my build env it uses samba from tools/depends/target/samba-gplv3

**Comment by MartijnKaijser** [comment]
@MrMC droid switched to GPLv3 a while ago

**Comment by MrMC** [comment]
Ouch, and google let Kodi in as GPLv3 code ? Amazing.

**Comment by mariuscampan** [comment]
I created a new pull request with the required patch for samba 3.6.12: https://github.com/xbmc/xbmc/pull/8944

**Comment by mariuscampan** [comment]
Beginner question: is this considered for Kodi 17?

**Comment by Rechi** [comment]
@mariuscampan have you tested this patch on top of recent master and does it still work?

**Comment by mariuscampan** [comment]
@Rechi unfortunately not. Will try to, but it might take a while as I do not have the setup anymore.

**Comment by mariuscampan** [comment]
@Rechi Works as intended on a linux build.

**Comment by Rechi** [comment]
Should the port be choosable in the `Add Network Location` dialog or should it just be a hidden feature?

**Comment by mariuscampan** [comment]
Right now it is kind of hidden - one must provide the port in the Server address field: <server>:<port>.

**Comment by Rechi** [comment]
The question is should it stay as hidden feature or should the port be more obvious selectable, like it is for http(s), dav(s), ftp, rss and sftp?

**Comment by wsnipex** [comment]
lets keep it "manual" as not many will ever need it

File: xbmc/filesystem/SMBFile.cpp [hunk 1]
```diff
@@ -232,6 +232,11 @@ std::string CSMB::URLEncode(const CURL &url)
   }
   flat += URLEncode(url.GetHostName());
 
+  if(url.HasPort())
```
Comment (Rechi) [inline xbmc/filesystem/SMBFile.cpp]: add a single white space between `if` and `(`

File: xbmc/filesystem/SMBFile.cpp [hunk 2]
```diff
@@ -232,6 +232,11 @@ std::string CSMB::URLEncode(const CURL &url)
...
+  {
+     flat += StringUtils::Format(":%i", url.GetPort());
```
Comment (Rechi) [inline xbmc/filesystem/SMBFile.cpp]: wrong indentation

**Review by Rechi — COMMENTED** [review]
Please fix the code style (https://codedocs.xyz/xbmc/xbmc/code_guidelines.html).

**Comment by Rechi** [comment]
@mariuscampan thanks for your contribution

**Comment by mariuscampan** [comment]
@Rechi Thanks for getting it in.

**Pull request merged**